lib/core/src/Cardano/Pool/Metrics.hs

```diff
@@ -195,15 +197,19 @@ monitorStakePools tr (block0, Quantity k) nl db@DBLayer{..} = do
         -> BlockHeader
         -> IO (FollowAction ErrMonitorStakePools)
     forward blocks nodeTip = handler $ do
-        (ep, dist) <- withExceptT ErrMonitorStakePoolsNetworkUnavailable $
-            stakeDistribution nl
+        let epochs = NE.nub $ view (#header . #slotId . #epochNumber) <$> blocks
+        distributions <- forM epochs $ \ep -> do
+            liftIO $ traceWith tr $ MsgStakeDistribution ep
+            withExceptT ErrMonitorStakePoolsNetworkUnavailable $
+                (ep,) <$> stakeDistribution nl ep
```

Comment:
So most likely we'll still make 1 request and 2 at epoch boundaries, without the tip changing — should be just fine.